### PR TITLE
Nk/driver alarm

### PR DESCRIPTION
**Pull request by nkrackow**
This PR changes the thermostat interlock to "alarm". Functionality is the same with the exception that the polarity of the critical topic is flipped (alarm=true is bad while interlocked=true was good)

### CODE REVIEW

File: src/hardware/driver/alarm.rs [hunk 1]
```diff
@@ -1,23 +1,23 @@
...
+///! The alarm is implemented via MQTT. See [Alarm] for details about the MQTT interface.
 use fugit::ExtU64;
 use miniconf::Miniconf;
 
 #[derive(Clone, Copy, Debug, Miniconf)]
-pub struct Interlock {
-    /// "Interlocked" topic. Publishing "true" onto this topic renews the interlock timeout.
-    /// Publishing "false" or failing to publish `true` for `timeout` trips the interlock.
+pub struct Alarm {
+    /// "Alarm" topic. Publishing "false" onto this topic renews the alarm timeout.
```
Comment (jordens) [inline src/hardware/driver/alarm.rs:10]:
```suggestion
    /// "Alarm" topic. Publish "false" onto this topic to indicate valid operating conditions. This renews the alarm timeout.
```

File: src/hardware/driver/alarm.rs [hunk 2]
```diff
@@ -1,23 +1,23 @@
...
+    /// Publishing "true" or failing to publish "false" for `timeout` trips the alarm.
```
Comment (jordens) [inline src/hardware/driver/alarm.rs:11]: And then does what?
What do I do to clear?

File: src/hardware/driver/alarm.rs [hunk 3]
```diff
@@ -49,29 +49,30 @@ impl Default for Interlock {
...
         match path {
-            Some("interlocked") => {
-                if self.interlocked && handle_is_some {
+            Some("alarm") => {
+                if !self.alarm && handle_is_some {
                     Some(Action::Reschedule(self.timeout.millis()))
-                } else if !self.interlocked && handle_is_some {
-                    Some(Action::Reschedule(0.millis()))
+                } else if self.alarm && handle_is_some {
+                    Some(Action::Trip(100.millis()))
```
Comment (jordens) [inline src/hardware/driver/alarm.rs:67]: Why 100?

File: src/bin/driver.rs [hunk 4]
```diff
@@ -83,12 +83,12 @@ pub struct Settings {
     /// See [StreamTarget#miniconf]
     stream_target: StreamTarget,
 
-    /// Thermostat interlock settings.
+    /// Thermostat alarm settings.
```
Comment (jordens) [inline src/bin/driver.rs:86]: I would not mention thermostat so prominently here and elsewhere. The feature is not limited to thermostat.
Do mention it as an example.

File: src/hardware/driver/mod.rs [hunk 5]
```diff
@@ -61,7 +61,8 @@ impl ChannelVariant {
 pub enum Reason {
     #[default]
     Reset, // Tripped after device reset
-    Thermostat,
+    ThermostatLimits,
```
Comment (jordens) [inline src/hardware/driver/mod.rs:64]: This isn't specific to `Thermostat` or `Limits`. Just call the two variants `Alarm` and `AlarmTimeout`.

**Review by jordens — CHANGES_REQUESTED** [review]

File: src/hardware/driver/alarm.rs (hunk 3 above)
Comment (nkrackow) [reply src/hardware/driver/alarm.rs:67]: oops, that was from debugging

**Review by jordens — APPROVED** [review]

**Pull request merged**